cmsplugin_profile/static/admin/js/profile_admin.js

```diff
@@ -27,11 +27,11 @@
             current_description = $("#id_description").val(),
             current_show_title = $("#id_show_title_on_thumbnails").prop('checked'),
 
-            has_unsaved_changes = profile_added || profile_deleted ||
+            has_unsaved_changes = save_rejected || profile_added ||
                 current_title != initial_title ||
                 current_description != initial_description ||
                 current_show_title != initial_show_title ||
-                profiles_have_changes();
+            profiles_have_changes();
```

Comment:
Indentation was better before. Please re-indent it or wrap in parentheses to make it clear that this won't execute if any of the above conditions is true.